src/main/scala/akka/persistence/cassandra/CassandraPluginConfig.scala

```diff
@@ -79,55 +83,76 @@ class CassandraPluginConfig(config: Config) {
       connectionPoolConfig.getInt("pool-timeout-millis")
     )
 
-  val clusterBuilder: Cluster.Builder = {
-    val b = Cluster.builder
-      .addContactPointsWithPorts(contactPoints.asJava)
-      .withPoolingOptions(poolingOptions)
-      .withQueryOptions(new QueryOptions().setFetchSize(fetchSize))
-    protocolVersion match {
-      case None    => b
-      case Some(v) => b.withProtocolVersion(v)
+  def contactPoints(system: ActorSystem): Future[Seq[InetSocketAddress]] = {
+    val contactPointsProviderClassName = config.getString("contact-points-provider")
+    if (contactPointsProviderClassName.isEmpty) {
```

Comment:
I would choose to have a separate config setting for enabling this functionality, but if this is conventional for Akka then fine of course.